app/ota_client.py

```diff
@@ -1002,15 +1002,15 @@ def _process_regular_file(self, rootfs_dir, target_dir, rfile_inf: RegularInf):
         main entry for paralleling processing regular files
         """
         try:
-            prev_inf = None
+            prev_inf: RegularInf = None
             # hardlinked file
             if int(rfile_inf.links) >= 2:
                 prev_inf = global_var_dict["tmp-dict-hardlink_reg"].setdefault(
                     rfile_inf.sha256hash, rfile_inf
                 )
                 # if the upcoming rfile entry is the first copy of hardlinked file
                 # then the prev_inf should be None
-                if prev_inf == rfile_inf:
+                if prev_inf and prev_inf.path == rfile_inf.path:
```

Comment:
I think `prev_inf` is always evaluated as True. I think `prev_inf.path == rfile_inf.path` is enough.

Reply:
Oh I should change it to `prev_inf is not None` instead.